crypto.go

```diff
@@ -10,29 +10,41 @@ import (
 	"golang.org/x/crypto/nacl/secretbox"
 )
 
-func keyedHash(key, text []byte) []byte {
+func keyedHash(key *[hashLen]byte, text []byte) *[hashLen]byte {
 	h := keyedHasher(key)
 	h.Write([]byte(text))
-	return h.Sum(nil)
+	var sum [hashLen]byte
+	r := h.Sum(sum[:0])
+	if &r[0] != &sum[0] {
```

Comment:
I don't see a new commit, but I'm not strongly against this implementation either. Any reason not to just check for equal lengths though?